packages/compass-databases-navigation/src/tree-item.tsx

```diff
@@ -58,28 +59,39 @@ export function useDefaultAction<T>(
 }
 
 const itemContainer = css({
-  display: 'flex',
-  alignItems: 'center',
   cursor: 'pointer',
   color: 'var(--item-color)',
-  backgroundColor: 'var(--item-bg-color)',
 
-  ':hover': {
+  '.item-action-controls': {
+    marginLeft: 'auto',
+    marginRight: spacing[1],
+  },
+
+  ':hover .item-background': {
+    display: 'block',
     backgroundColor: 'var(--item-bg-color-hover)',
   },
 
+  '.item-action-controls:hover + .item-background': {
+    display: 'none',
+  },
```

Comment:
One super minor thing, so totally feel free to ignore: something that tripped me up during a review is that emotion seem to have different way to concatenate selectors based on whether it is a pseudo selector or not (so like `:hover` will be `.as6rf:hover`, but `.item` will become `.as6rf .item`) and so maybe just to make this explicit we can use `&` there

```suggestion
  '& .item-action-controls': {
    marginLeft: 'auto',
    marginRight: spacing[1],
  },

  '&:hover .item-background': {
    display: 'block',
    backgroundColor: 'var(--item-bg-color-hover)',
  },

  '& .item-action-controls:hover + .item-background': {
    display: 'none',
  },
```

Reply:
Seems sensible. I was surprised it worked this way actually, but saw it done elsewhere which is why I did it this way. I'll add the &